### PR TITLE
Feature/gh 11450 review single selection bundle

### LINKED ISSUE #11450: Review: Single selection bundle, strict mode issues, comments, method signatures

### PR DESCRIPTION
**Pull request by ChristophHi**

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/components/attribute/product-card/configurator-attribute-product-card.component.ts [hunk 1]
```diff
@@ -127,6 +107,10 @@ export class ConfiguratorAttributeProductCardComponent implements OnInit {
     return isProductCardSelected ? isProductCardSelected : false;
   }
 
+  /**
+   * Computes product price
```
Comment (schuppell) [inline feature-libs/product-configurator/rulebased/components/attribute/product-card/configurator-attribute-product-card.component.ts]: Could you elaborate more which price is taken in which case? This is not clear with this description.

File: feature-libs/product-configurator/rulebased/components/attribute/quantity/configurator-attribute-quantity.service.ts [hunk 2]
```diff
@@ -23,8 +23,8 @@ export class ConfiguratorAttributeQuantityService {
    * @returns Render a quantity component?
    */
   withQuantity(
-    dataType: Configurator.DataType,
-    uiType: Configurator.UiType
+    dataType: Configurator.DataType | undefined,
+    uiType: Configurator.UiType | undefined
```
Comment (schuppell) [inline feature-libs/product-configurator/rulebased/components/attribute/quantity/configurator-attribute-quantity.service.ts]: Could you add a unit test where you call this method with undefined parameters and check wether or not the method behaves as intended.

File: feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle/configurator-attribute-single-selection-bundle.component.spec.ts [hunk 3]
```diff
@@ -1,20 +1,21 @@
...
-import { ConfiguratorAttributeSingleSelectionBundleComponent } from './configurator-attribute-single-selection-bundle.component';
-import { ConfiguratorShowMoreComponent } from '../../../show-more/configurator-show-more.component';
-import { I18nTestingModule } from '@spartacus/core';
-import { ItemCounterComponent } from '@spartacus/storefront';
 import { ReactiveFormsModule } from '@angular/forms';
 import { RouterTestingModule } from '@angular/router/testing';
+import { I18nTestingModule } from '@spartacus/core';
+import { CommonConfiguratorTestUtilsService } from '@spartacus/product-configurator/common';
+import { ItemCounterComponent } from '@spartacus/storefront';
 import { UrlTestingModule } from 'projects/core/src/routing/configurable-routes/url-translation/testing/url-testing.module';
```
Comment (schuppell) [inline feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle/configurator-attribute-single-selection-bundle.component.spec.ts]: Not from your PR, but nevertheless this path is wrong and should not be specifified in an absolute way.

Comment (schuppell) [inline feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle/configurator-attribute-single-selection-bundle.component.spec.ts]: Not from your PR, but nevertheless this path is wrong and should not be specified in an absolute way.

**Review by schuppell — CHANGES_REQUESTED** [review]
See comments

File: feature-libs/product-configurator/rulebased/components/attribute/product-card/configurator-attribute-product-card.component.ts (hunk 1 above)
Comment (ChristophHi) [reply feature-libs/product-configurator/rulebased/components/attribute/product-card/configurator-attribute-product-card.component.ts]: Good point. I saw that the method is named a bit weirdly, as its purpose is boolean => changed that, adjusted the signature, tests and comments

File: feature-libs/product-configurator/rulebased/components/attribute/quantity/configurator-attribute-quantity.service.ts (hunk 2 above)
Comment (ChristophHi) [reply feature-libs/product-configurator/rulebased/components/attribute/quantity/configurator-attribute-quantity.service.ts]: Not needed, as I could remove `| undefined` here

**Review by schuppell — APPROVED** [review]
Looks good.

**Pull request merged**